rct229/rulesets/ashrae9012019/section19/section19rule13.py

```diff
@@ -0,0 +1,255 @@
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.std_comparisons import std_equal
+
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_1,
+    HVAC_SYS.SYS_2,
+    HVAC_SYS.SYS_3,
+    HVAC_SYS.SYS_4,
+    HVAC_SYS.SYS_5,
+    HVAC_SYS.SYS_6,
+    HVAC_SYS.SYS_7,
+    HVAC_SYS.SYS_8,
+    HVAC_SYS.SYS_11,
+    HVAC_SYS.SYS_12,
+    HVAC_SYS.SYS_13,
+]
+
+LIGHTING_SPACE = schema_enums["LightingSpaceOptions2019ASHRAE901TG37"]
+TEMP_DELTA_17F = 17.0 * ureg("R")
+TEMP_DELTA_20F = 20.0 * ureg("R")
+
+
+class Section19Rule13(RuleDefinitionListIndexedBase):
+    """Rule 13 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule13, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section19Rule13.HVACRule(),
+            index_rmr="baseline",
+            id="19-13",
+            description="For baseline system types 1-8 and 11-13, system design supply airflow rates shall be based on a supply-air-to-room temperature set-point difference of 20°F or the minimum outdoor airflow rate, or the airflow rate required to comply with applicable codes or accreditation standards, whichever is greater. For systems with multiple zone thermostat setpoints, use the design set point that will result in the lowest supply air cooling set point or highest supply air heating set point.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.8.1 and Exception 1",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        dict_of_zones_and_terminal_units_served_by_hvac_sys_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
+        )
+
+        zone_info = {}
+        for hvac_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b:
+            for zone_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
+                hvac_id_b
+            ]["zone_list"]:
+                zone_info[zone_id_b] = {}
+                zone_info[zone_id_b][
+                    "are_all_hvac_sys_fan_objs_autosized"
+                ] = are_all_hvac_sys_fan_objs_autosized(rmi_b, hvac_id_b)
+                zone_info[zone_id_b]["proposed_supply_airflow"] = ZERO.FLOW
+                zone_info[zone_id_b][
+                    "all_design_setpoints_delta_Ts_are_per_reqs"
+                ] = ZERO.TEMPERATURE
+
+                # get `design_thermostat_cooling_setpoint` and `design_thermostat_heating_setpoint`
+                for zone_b in find_all(
+                    '$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")]',
+                    rmi_b,
+                ):
```

Comment:
The logic seems wrong here.

The RDS is trying to calculate the HVAC design thermostate setpoint for heating and cooling based highest and lowest zone_design_thermostat_setpoints respectively. Based on the implementation, I will just include the zone_design_thermostat_cooling_setpoint & zone_design_thermostat_heating_setpoint in the `create_data` function.

```python
zone_info[zone_id_b]["design_thermostat_cooling_setpoint"] = min(find_one(f'$.buildings[*].building_segments[*].zones[(@.id == "{zone_id_b}")].design_thermostat_cooling_setpoint',rmi_b),
100*ureg("C")) 
# set to some very large temperature. - maybe create an infinity, I am not sure how pint could handle infinity, you can try this out.
zone_info[zone_id_b]["design_thermostat_heating_setpoint"] = min(find_one(f'$.buildings[*].building_segments[*].zones[(@.id == "{zone_id_b}")].design_thermostat_heating_setpoint',rmi_b),
ZERO.temperature)
```
You should also remove the for loop - instead, use `find_one` function.
The above should do the trick to extract the design thermostat.

Reply:
Addressed.